extensions/default/src/MergeDataSource/index.js

```diff
@@ -0,0 +1,195 @@
+   */
+  const callForAllDataSourcesAsync = async (path, args) => {
+    const mergeMap = {
+      'query.studies.search': {
+        mergeKey: 'studyInstanceUid',
+        level: 0,
+        /** TODO */
+        tagFunc: x => x,
+      },
+      'retrieve.series.metadata': {
+        mergeKey: 'seriesInstanceUid',
+        level: 0,
+        tagFunc: (series, sourceName) => {
+          series.forEach(series => {
+            series.RetrieveAETitle = sourceName;
+            DicomMetadataStore.updateSeriesMetadata(series);
+          });
+          return series;
+        },
+      },
+    };
+
+    const { mergeKey, level, tagFunc } = mergeMap[path] || { tagFunc: x => x, level: -1 };
+
+    const dataSourceDefs = Object.values(extensionManager.dataSourceDefs);
+    const promises = [];
+    let globalData = [];
+
+    for (const dataSourceDef of dataSourceDefs) {
+      const { configuration, sourceName } = dataSourceDef;
+      if (configuration && dataSourceNames.includes(sourceName)) {
+        const [dataSource] = extensionManager.getDataSources(sourceName);
+        const promise = executeFunction(dataSource, path, args);
+        promises.push(promise);
+
+        promise.then(data =>
+          processDedupedData({
+            sourceName,
+            globalData,
+            incomingData: data,
+            mergeKey,
+            level,
+            tagFunc,
+          })
+        );
+      }
+    }
+
+    await Promise.allSettled(promises);
+
+    return globalData.flat();
+  };
+
+  /**
+   * Calls a function with the given path and arguments on all data sources that are included in the given list of data source names.
+   * Returns an array of the results from each data source.
+   *
+   * @param {string} path - The path of the function to call on each data source.
+   * @param {Array} args - The arguments to pass to the function on each data source.
+   * @returns {Array} - An array of the results from each data source.
+   */
+  const callForAllDataSources = (path, args) => {
+    const dataSourceDefs = Object.values(extensionManager.dataSourceDefs);
+    const globalData = [];
+    for (const dataSourceDef of dataSourceDefs) {
+      const { configuration, sourceName } = dataSourceDef;
+      if (configuration && dataSourceNames.includes(sourceName)) {
+        const [dataSource] = extensionManager.getDataSources(sourceName);
+        const data = executeFunction(dataSource, path, args);
+        globalData.push(data);
+      }
+    }
+    return globalData.flat();
+  };
+
+  /**
+   * Calls the default data source with the given path and arguments.
+   * @param {string} path - The path to the function to be executed.
+   * @param {Array} args - The arguments to be passed to the function.
+   * @returns {*} - The result of executing the function.
+   */
+  const callForDefaultDataSource = (path, args) => {
+    const defaultDataSourceName = dataSourceNames[0];
```

Comment:
This is NOT the default - the default is specifically specified by name as a parameter 'defaultDataSourceName' or some such value.  Also, the default might be specific to the merge configuration, so I think you should have two levels of default to allow the merge configuration to specify a specific default.

Reply:
Updated, thanks!